### PR TITLE
Typescript, Stats API v2 playground

### PR DESCRIPTION
**Pull request by macobo**
This PR follows up on https://github.com/plausible/docs/pull/532 by:
1. Adding typescript to our docs
2. Adding a new "Stats API v2 playground page"

image

The UI/UX on this is pretty anemic - would gladly accept volunteers to help improve it :)

### CODE REVIEW

**Comment by apata** [comment]
>  The UI/UX on this is pretty anemic - would gladly accept volunteers to help improve it :)
I could polish the styling and UX a little bit. I remember you mentioned Stripe as inspiration (https://docs.stripe.com/api?lang=node), and I think I could make the editor and buttons look a bit more rounded like on that page.

File: src/css/custom.css [hunk 1]
```diff
@@ -46,3 +46,32 @@
...
+  font-weight: 500;
+  line-height: 20px;
+  list-style: none;
+  margin: 0;
+  padding: 10px 12px;
+  text-align: center;
+  transition: all 200ms;
+  vertical-align: baseline;
+  touch-action: manipulation;
+}
```
Comment (apata) [inline src/css/custom.css:77]: **nitpick**: This needs more contrast in dark mode. 
```suggestion
  background-color: var(--ifm-background-surface-color);
  border-radius: var(--ifm-global-radius);
  border-width: 0;
  color: var(--ifm-navbar-link-color);
  cursor: pointer;
  font-family: "Haas Grot Text R Web", "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-size: 16px;
  font-weight: 500;
  line-height: 20px;
  list-style: none;
  margin: 0;
  padding: 10px 12px;
  text-align: center;
  transition: all 200ms;
  vertical-align: baseline;
  touch-action: manipulation;
}

.playground-run-code button:hover {
  background-color: var(--ifm-color-emphasis-200);
}
```

**Comment by macobo** [comment]
> I could polish the styling and UX a little bit. I remember you mentioned Stripe as inspiration (https://docs.stripe.com/api?lang=node), and I think I could make the editor and buttons look a bit more rounded like on that page.

Feel free to take a stab at it and push onto this branch! :)

**Review by ukutaht — APPROVED** [review]

File: src/css/custom.css (hunk 1 above)
Comment (macobo) [reply src/css/custom.css]: Good feedback. Changed the color a bit (-0 is just white) but this works well.

**Pull request merged**